rule/rule.go

```diff
@@ -115,24 +130,40 @@ func LoadWorkspaceData(path, pkg string, data []byte) (*File, error) {
 	if err != nil {
 		return nil, err
 	}
-	return ScanAST(pkg, ast), nil
+	return ScanAST(pkg, ast, ast.Stmt), nil
+}
+
+// LoadMacroData is similar to LoadData but parses the data as a
+// bzl file.
+func LoadMacroData(path, pkg, defName string, data []byte) (*File, error) {
+	ast, err := bzl.ParseDefault(path, data)
+	if err != nil {
+		return nil, err
+	}
+	defStmt := scanForDef(ast.Stmt, defName)
+	if defStmt == nil {
+		return nil, fmt.Errorf("Could not find function name in .bzl file %s", defName)
+	}
+	f := ScanAST(pkg, ast, defStmt.Body)
+	f.function = defStmt
+	return f, nil
 }
 
 // ScanAST creates a File wrapped around the given syntax tree. This tree
 // will be modified by editing methods.
-func ScanAST(pkg string, bzlFile *bzl.File) *File {
+func ScanAST(pkg string, bzlFile *bzl.File, stmt []bzl.Expr) *File {
```

Comment:
I'd rather not make a breaking API change if we don't have to. Let's add a function, `ScanASTBody` with the new signature. `ScanAST` can call that with `f.File.Stmt`.

Reply:
Fixed by creating `ScanASTBody` and updating `LoadMacroData` and